### PR TITLE
Script linux fix

### PR DESCRIPTION
**Pull request by nghuyenthevinh2000**
Closes: #XXX

## Context and purpose of the change

The script to setup a network is very inconsistent. This is an attempt to fix this inconsistency.

The exact case that this fix for is make start-docker

Screenshot 2022-11-02 at 01 30 10

## Brief Changelog

- Change to go install to GOBIN instead of go build (which is very inconsistent).
- Improve in scripts/vars so that people can add admin mnemonic.
- Change fixed address to STRIDE_ADMIN_ADDRESS.
- Change to docker-compose

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [ ] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: utils/admins.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 package utils
 
 var Admins = map[string]bool{
-	"stride1k8c2m5cn322akk5wy8lpt87dd2f4yh9azg7jlh": true, // F5
+	"stride1u20df3trc2c2zdhm8qvh2hdjx9ewh00sv6eyy8": true, // F5
```
Comment (asalzmann) [inline utils/admins.go:4]: This address should revert to `stride1k8c2m5cn322akk5wy8lpt87dd2f4yh9azg7jlh` when `make stop-docker` is run

File: scripts/vars.sh [hunk 2]
```diff
@@ -79,7 +79,8 @@ STRIDE_VAL_PREFIX=val
 STRIDE_DENOM=$STRD_DENOM
 STRIDE_RPC_PORT=26657
 STRIDE_ADMIN_ACCT=admin
-STRIDE_ADMIN_ADDRESS=stride1u20df3trc2c2zdhm8qvh2hdjx9ewh00sv6eyy8
+STRIDE_ADMIN_ADDRESS=stride1aj7yqqpxnrxwva4eh2w7u4mgsvfjl937v4d89s
+STRIDE_ADMIN_MNEMONIC="pink goose diesel cool private green aware dog coral fun coach prefer spice sword car gadget melt wish trust twelve saddle brain page man"
```
Comment (sampocs) [inline scripts/vars.sh:83]: Can you set these to the following?

STRIDE_ADMIN_ADDRESS=stride1u20df3trc2c2zdhm8qvh2hdjx9ewh00sv6eyy8
STRIDE_ADMIN_MNEMONIC="tone cause tribe this switch near host damage idle fragile antique tail soda alien depth write wool they rapid unfold body scan pledge soft"

File: scripts/upgrades/Dockerfile.cosmovisor [hunk 3]
```diff
@@ -2,6 +2,7 @@
 FROM golang:1.18-alpine3.15 AS builder
 
 ARG old_commit_hash
+ARG STRIDE_ADMIN_ADDRESS
```
Comment (sampocs) [inline scripts/upgrades/Dockerfile.cosmovisor:5]: nit: can you either set these both to uppercase or both to lowercase to keep the styling consistent?

File: scripts/local-to-mainnet/commands.sh [hunk 4]
```diff
@@ -31,13 +31,13 @@ docker-compose logs -f relayer-host | sed -r -u "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2}
 
 #### REGISTER HOST
 # IBC Transfer from HOST to stride (from relayer account)
-build/osmosisd tx ibc-transfer transfer transfer $transfer_channel stride1u20df3trc2c2zdhm8qvh2hdjx9ewh00sv6eyy8 4000000uosmo --from hot --chain-id osmosis-1 -y --keyring-backend test --node http://osmo-fleet-direct.main.stridenet.co:26657
+build/osmosisd tx ibc-transfer transfer transfer $transfer_channel $STRIDE_ADMIN_ADDRESS 4000000uosmo --from hot --chain-id osmosis-1 -y --keyring-backend test --node http://osmo-fleet-direct.main.stridenet.co:26657
```
Comment (sampocs) [inline scripts/local-to-mainnet/commands.sh:34]: these scripts in `local-to-mainnet` are actually not related to dockernet - can you revert these? (same thing in `templates/commands.sh`)

File: docker-compose.yml [hunk 5]
```diff
@@ -2,183 +2,236 @@ version: "3.8"
 services:
   stride1:
     image: stridezone:stride
+    user: root
     volumes:
       - ./scripts/state/stride1:/home/stride/.stride
+    command: strided start --home /home/stride/.stride
```
Comment (sampocs) [inline docker-compose.yml:8]: I don't think these commands are necessary since they're in the dockerfile

File: docker-compose.yml [hunk 6]
```diff
@@ -2,183 +2,236 @@ version: "3.8"
 services:
   stride1:
     image: stridezone:stride
+    user: root
```
Comment (sampocs) [inline docker-compose.yml:5]: What's the reason for running as root?

**Review by sampocs — COMMENTED** [review]
Thanks for making these changes! A few small comments below:

File: docker-compose.yml (hunk 6 above)
Comment (nghuyenthevinh2000) [reply docker-compose.yml:5]: when volume is mounted, docker cannot access mounted files due to access rights

File: docker-compose.yml (hunk 5 above)
Comment (nghuyenthevinh2000) [reply docker-compose.yml:8]: in dockerfile, home is pointed to /root/.stride by default so the docker fails. Best would be to change this in Dockerfile CMD.

**Comment by sampocs** [comment]
@nghuyenthevinh2000 Hey can you give the `linux-dockernet` branch a try (#327)? I incorporated your changes from this branch, as well as some others, and tested it on Debian and Ubuntu VMs

**Pull request closed without merging**